extensions/sql-database-projects/src/extension.ts

```diff
@@ -22,7 +22,7 @@ export function activate(context: vscode.ExtensionContext): Promise<SqlDatabaseP
 	context.subscriptions.push(TelemetryReporter);
 
 	// Register the Sql project task provider
-	const workspaceFolders = vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0 ? vscode.workspace.workspaceFolders : undefined;
+	const workspaceFolders = vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders?.length > 0 ? vscode.workspace.workspaceFolders : undefined;
```

Comment:
I was just mentioning that `vscode.workspace.workspaceFolders` isn't needed anymore now that you have ?

Reply:
yes, it shouldn't as we are using the optional chaining, but I believe typescript is validating it differently compared to null-condition operator (like in c#). It seems we are checking the same thing twice, but since the workspaceFolders can return a list | undefined, first check is required to verify if the folders exists then only go for second check to get the lenght, otherwise it could check for undefined?.length which is causing the runtime error 
"'vscode.workspace.workspaceFolders.length' is possibly 'undefined'.ts(18048)"